avocado/utils/network/interfaces.py

```diff
@@ -128,6 +128,43 @@ def add_ipaddr(self, ipaddr, netmask):
         except Exception as ex:
             raise NWException("Failed to add address {}".format(ex))
 
+    def add_vlan_tag(self, vlan_num):
+        """Configure 802.1Q VLAN tagging to the interface
+
+        This method will try to add a VLAN tag to this interface and return the
+        VLAN NetworkInterface object. If it fails, the method will raise a
+        NWException.
+
+        :param vlan_num: VLAN ID
+        :return: NetworkInterface object representing the newly created VLAN interface
+        """
+
+        vlan_name = "{}.{}".format(self.name, vlan_num)
+        cmd = "ip link add link {} name {} type vlan id {}".format(self.name, vlan_name,
+                                                                   vlan_num)
+        try:
+            run_command(cmd, self.host, sudo=True)
+            return NetworkInterface(vlan_name, self.host)
```

Comment:
Any reason for creating a new object here? 

Since developers will be manipulating the self object already I think this new object is not needed. We could do something like this:


```python
def add_vlan_tag(self, vlan_num, vlan_name=None):
  vlan_name is vlan_name or "{}.{}".format(self.name, vlan_num) 
  cmd = "ip link add link {} name {} type vlan id {}".format(self.name,
                                                             vlan_name,
                                                             vlan_num)
  try:
    run_command(cmd, self.host, sudo=True)
  except Exception as ex:
    raise NWException("Failed to add VLAN tag: {}".format(ex))
```

And add methods to check/return the vlans, something like:

```python
@property
def vlans(self):
   ...
   return list_of_vlan_ids
```